### ISSUE TITLE
Loading image programmatically with transformation is inconsistent 

### ISSUE THREAD

**Issue by CarlosEsco** [coil-kt/coil#640]
I am using contour from cash app to programmatically layout some of my screens.  I have an image defined
```
   val cover = ImageView(context).apply {
        elevation = 10f
        translationZ = 10f
        this.loadAny(url){
            transformations(RoundedCornersTransformation(20f))
        }
    }
```

as you can see it has a transformation of rounded corners of 20f
Here is it being laid out
```
 cover.layoutBy(
            x = leftTo { parent.left() + 8.xdip }.rightTo { parent.right() - 8.xdip },
            y = topTo { parent.top() + 16.ydip }.bottomTo { card.top() - 8.ydip }
        )
```

I have a bunch of preview thumbnails that upon clicking replace the loaded image with the new cover
```
 private fun onCoverClicked(position: Int) {
        cover.load(urls[position]) {
            transformations(RoundedCornersTransformation(20f))
        }
    }
```
upon doing this.  The image now seems to fill the original imageview. 
original load
![Screenshot_20210118-215140_Neko](https://user-images.githubusercontent.com/2092019/104982047-d0bfbf00-59d7-11eb-8f21-98664cfd1775.jpg)
after click
![Screenshot_20210118-214805_Neko](https://user-images.githubusercontent.com/2092019/104982081-e0d79e80-59d7-11eb-91ac-444645a56a93.jpg)

Then to add to the issue.  When i remove transformations it loads the image correctly as well as the initial image is in the correct ratios
![Screenshot_20210118-215346_Neko](https://user-images.githubusercontent.com/2092019/104982130-049ae480-59d8-11eb-804b-77ae497552d7.jpg)

If I use  jp.wasabeef.transformers.coil.RoundedCornersTransformation  everything works as expected
![Screenshot_20210118-215227_Neko](https://user-images.githubusercontent.com/2092019/104982166-14b2c400-59d8-11eb-8719-af0e1ae6ac10.jpg)

So it seems the default image loading when a transformation is applied is causing inconsistent behavior

**Comment by colinrtwhite** [comment]
It looks like both images were cropped to fill the target, which is the [intended behaviour](https://github.com/coil-kt/coil/blob/master/coil-base/src/main/java/coil/transform/RoundedCornersTransformation.kt#L26). You may be able to use `Scale.FIT` to have the image fit the target instead of fill.

**Comment by daksh7011** [comment]
I was having same issue with dynamic bounds. Have a look [here](https://github.com/coil-kt/coil/issues/421#issuecomment-640133205).

**Comment by CarlosEsco** [comment]
I agree it looks like it cropped to fill, however shouldn't the first load do the same, for consistency?

**Comment by colinrtwhite** [comment]
@CarlosEsco It will crop to the size of the target which may change between requests. You can set a breakpoint in [`RoundedCornersTransformation` ](https://github.com/coil-kt/coil/blob/master/coil-base/src/main/java/coil/transform/RoundedCornersTransformation.kt#L49) to see what size it is being cropped to. It's tough to say what the exact issue is without a sample project that reproduces the issue.

**Comment by CarlosEsco** [comment]
Here is a sample project
https://github.com/CarlosEsco/coil

**Comment by colinrtwhite** [comment]
@CarlosEsco Great, thanks. I'll take a look.

**Comment by colinrtwhite** [comment]
This is fixed in Coil 2.0.0.
